app/src/main/java/org/mozilla/focus/tiles/HomeTilesManager.kt

```diff
@@ -47,4 +60,14 @@ object HomeTilesManager {
     private fun getHomeTilesPreferences(context: Context): SharedPreferences {
         return context.getSharedPreferences(HOME_TILES_PREFS, MODE_PRIVATE)
     }
+
+    fun getCustomHomeTilesList(context: Context): List<HomeTile> {
+        val sharedPreferences = context.getSharedPreferences(HOME_TILES_PREFS, MODE_PRIVATE)
+        val sitesJSONArray = getCustomSitesJSONArray(sharedPreferences)
+        val homeTiles = mutableListOf<HomeTile>()
+        for (i in sitesJSONArray.length() - 1 downTo 0) {
```

Comment:
nit: fwiw, you could add a `JSONArray.map` extension fn, which would make this more readable, though this is pretty obnoxious since you have to deal with generics. I actually wrote a flatMapObj fn in another WIP: https://github.com/mcomella/firefox-tv/commit/72a4db66ef68d59afb6a4744ba6f02bfe689c713